source/ScreensaverHero.hx

```diff
@@ -0,0 +1,61 @@
+package;
+
+import flixel.FlxG;
+import flixel.FlxSprite;
+import flixel.math.FlxPoint;
+import flixel.math.FlxRandom;
+import flixel.util.FlxColor;
+
+class ScreensaverHero extends FlxSprite
+{
+	var randomSeed:FlxRandom;
+	var xDirectionIndicator:Int;
+	var yDirectionIndicator:Int;
+
+	public function new(X:Float = 0, Y:Float = 0)
+	{
+		super(X, Y);
+		this.makeGraphic(40, 20, FlxColor.BLUE, false);
+		xDirectionIndicator = Std.int(FlxG.random.int(-1, 1));
+		yDirectionIndicator = Std.int(FlxG.random.int(-1, 1));
+
+		if (xDirectionIndicator == 0 && yDirectionIndicator == 0)
+		{
+			// now using awesome built in number generator!
+			xDirectionIndicator = FlxG.random.sign(50);
+			yDirectionIndicator = FlxG.random.sign(50);
+		}
+	}
+
+	override public function update(elapsed:Float)
```

Comment:
The ideas I mentioned earlier about simplifying `create()` by breaking functionality into smaller, more focused private methods applies to stuff like update too - this class has `color changing`, `position updating`, `direction changing`, and `screen wrapping` functionality all hanging out side-by-side, which can make it more difficult to read over and update when you come back to this project later.